plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java

```diff
@@ -235,8 +239,13 @@ public RedshiftClient(
             RemoteQueryModifier queryModifier)
     {
         super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, queryModifier, true);
-        ConnectorExpressionRewriter<ParameterizedExpression> connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
+        connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
                 .addStandardRules(this::quoted)
+                .add(new RewriteComparison(ImmutableSet.of(ComparisonOperator.EQUAL, ComparisonOperator.NOT_EQUAL)))
+                .map("$less_than(left, right)").to("left < right")
+                .map("$less_than_or_equal(left, right)").to("left <= right")
+                .map("$greater_than(left, right)").to("left > right")
+                .map("$greater_than_or_equal(left, right)").to("left >= right")
```

Comment:
This would apply for all the operators like double and real as well - Don't we need to Handle NaN for the same ? I think we also need to consider on how comparison is applied on NaN and Infinity values

Reply:
we have similar rules in PostgreSQL for numeric types: `"tinyint", "smallint", "integer", "bigint", "decimal", "real", "double"`

Reply:
We could try something like that to start with.